homeassistant/components/romy/sensor.py

```diff
@@ -0,0 +1,112 @@
+    UnitOfLength,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import RomyVacuumCoordinator
+from .entity import RomyEntity
+
+SENSORS: list[SensorEntityDescription] = [
+    SensorEntityDescription(
+        key="battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="rssi",
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="dustbin_sensor",
+        translation_key="dustbin_sensor",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="total_cleaning_time",
+        translation_key="total_cleaning_time",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="total_number_of_cleaning_runs",
+        translation_key="total_number_of_cleaning_runs",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement="runs",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="total_area_cleaned",
+        translation_key="total_area_cleaned",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=AREA_SQUARE_METERS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="total_distance_driven",
+        translation_key="total_distance_driven",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfLength.METERS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ROMY vacuum cleaner."""
+
+    coordinator: RomyVacuumCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        RomySensor(coordinator, entity_description)
+        for entity_description in SENSORS
+        if entity_description.key in coordinator.romy.sensors
+    )
+
+
+class RomySensor(RomyEntity, SensorEntity):
+    """RomySensor Class."""
+
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: RomyVacuumCoordinator,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize ROMYs StatusSensor."""
```

Comment:
Stale docstring

Reply:
changed to
"""Initialize the RomySensor."""
and
"""Initialize the RomyBinarySensor."""

whats missing or what do you prefer?

Reply:
I just noticed that it isnt a StatusSensor, so that was my issue mostly